### PR TITLE
Added ped model data.

### PR DESCRIPTION
**Pull request by aa15032261**
image

### CODE REVIEW

**Comment by howdoiusekeyboard** [comment]
Can you add preview option in this?

**Comment by aa15032261** [comment]
> Can you add preview option in this?

Maybe in a future update

**Comment by howdoiusekeyboard** [comment]
Add an option to be able to auto give all weapons to appropriate ped models when changing ped model

**Comment by aa15032261** [comment]
> Add an option to be able to spawn appropriate ped types with weapons?

Like body guards?

File: BigBaseV2/src/api/remote.hpp [hunk 1]
```diff
@@ -49,6 +49,8 @@ namespace big::remote
 			std::string local_etag = "";
 			std::string remote_etag = "";
 
+			std::vector<std::string> headers = { "User-Agent: Mozilla/5.0 (Windows; U; Windows NT 6.1; en-US) AppleWebKit/533.2 (KHTML, like Gecko) Chrome/6.0" };
```
Comment (Yimura) [inline BigBaseV2/src/api/remote.hpp]: Use `constexpr std::vector<std::string>`  the value is never modified and the compiler can optimize this to not have to construct the vector each time.

File: BigBaseV2/src/main.cpp [hunk 2]
```diff
@@ -76,6 +77,7 @@ BOOL APIENTRY DllMain(HMODULE hmod, DWORD reason, PVOID)
 					auto notification_service_instance = std::make_unique<notification_service>();
 					auto pickup_service_instance = std::make_unique<pickup_service>();
 					auto player_service_instance = std::make_unique<player_service>();
+					auto gta_data_service_instance = std::make_unique<gta_data_service>();
```
Comment (Yimura) [inline BigBaseV2/src/main.cpp:80]: Call reset for this instance when unloading the DLL, see the other services that are beind called with .reset() further down.

File: BigBaseV2/src/services/gta_data/gta_data_service.hpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+	class ped_item {
+	public:
+		ped_item();
+		ped_item(nlohmann::json& item_json);
+
+		std::string name;
+		std::string ped_type;
+		Hash hash;
+	};
```
Comment (Yimura) [inline BigBaseV2/src/services/gta_data/gta_data_service.hpp]: These classes could be moved to their own files, see the player service.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Comment by howdoiusekeyboard** [comment]
Can you filter ped type as per https://alloc8or.re/gta5/doc/enums/ePedType.txt

**Comment by aa15032261** [comment]
> Can you filter ped type as per https://alloc8or.re/gta5/doc/enums/ePedType.txt

There's a filter above model name textbox.

**Comment by howdoiusekeyboard** [comment]
Yeah I see that, thats not what I meant
Ped types in this are filtered as: ['Animal', 'civmale', 'CIVFEMALE', 'COP', 'PLAYER_1', 'PLAYER_2', 'PLAYER_0', 'army', 'MEDIC', 'FIREMAN', 'Swat']
Its better to filter it as per ePedType https://alloc8or.re/gta5/doc/enums/ePedType.txt which is there in enums

**Comment by aa15032261** [comment]
> Yeah I see that, thats not what I meant Ped types in this are filtered as: ['Animal', 'civmale', 'CIVFEMALE', 'COP', 'PLAYER_1', 'PLAYER_2', 'PLAYER_0', 'army', 'MEDIC', 'FIREMAN', 'Swat'] Its better to filter it as per ePedType https://alloc8or.re/gta5/doc/enums/ePedType.txt which is there in enums

Some of the ped types aren't used by the models, at least according to the database. But I think I can add more categories in the future based on other info.

**Review by Yimura — APPROVED** [review]

**Pull request merged**